src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1660,6 +1665,21 @@ private static HttpClient CreateHttpClient(HttpClientParameters parameters, Http
 
                     try
                     {
+                        if (parameters.ServicePoint is not null)
```

Comment:
Nit: You can do something like
```c#
if (parameters.ServicePoint is { } servicePoint)

if (servicePoint.KeepAlive is { } keepAlive)
```

to avoid repeating the same `parameters.ServicePoint.KeepAlive.` expression.